lightning/src/ln/channel.rs

```diff
@@ -2686,21 +2689,121 @@ impl<Signer: WriteableEcdsaChannelSigner> Channel<Signer> {
+			if capacity_minus_commitment_fee_msat < (real_dust_limit_timeout_sat as i64) * 1000 {
+				let one_htlc_difference_msat = max_reserved_commit_tx_fee_msat - min_reserved_commit_tx_fee_msat;
+				debug_assert!(one_htlc_difference_msat != 0);
+				capacity_minus_commitment_fee_msat += one_htlc_difference_msat as i64;
+				capacity_minus_commitment_fee_msat = cmp::min(real_dust_limit_timeout_sat as i64 * 1000 - 1, capacity_minus_commitment_fee_msat);
+				available_capacity_msat = cmp::max(0, cmp::min(capacity_minus_commitment_fee_msat, available_capacity_msat as i64)) as u64;
+			} else {
+				available_capacity_msat = capacity_minus_commitment_fee_msat as u64;
+			}
+		} else {
+			// If the channel is inbound (i.e. counterparty pays the fee), we need to make sure
+			// sending a new HTLC won't reduce their balance below our dust threshold.
+			let mut real_dust_limit_success_sat = self.counterparty_dust_limit_satoshis;
+			if !self.opt_anchors() {
+				real_dust_limit_success_sat += self.feerate_per_kw as u64 * htlc_success_tx_weight(false) / 1000;
+			}
+
+			let htlc_above_dust = HTLCCandidate::new(real_dust_limit_success_sat * 1000, HTLCInitiator::LocalOffered);
+			let max_reserved_commit_tx_fee_msat = self.next_remote_commit_tx_fee_msat(htlc_above_dust, None);
+
+			let holder_selected_chan_reserve_msat = self.holder_selected_channel_reserve_satoshis * 1000;
+			let remote_balance_msat = (self.channel_value_satoshis * 1000 - self.value_to_self_msat)
+				.saturating_sub(inbound_stats.pending_htlcs_value_msat);
+
+			if remote_balance_msat < max_reserved_commit_tx_fee_msat + holder_selected_chan_reserve_msat {
+				// If another HTLC's fee would reduce the remote's balance below the reserve limit
+				// we've selected for them, we can only send dust HTLCs.
+				available_capacity_msat = cmp::min(available_capacity_msat, real_dust_limit_success_sat * 1000 - 1);
+			}
+		}
+
+		let mut next_outbound_htlc_minimum_msat = self.counterparty_htlc_minimum_msat;
+
+		// If we get close to our maximum dust exposure, we end up in a situation where we can send
+		// between zero and the remaining dust exposure limit remaining OR above the dust limit.
+		// Because we cannot express this as a simple min/max, we prefer to tell the user they can
+		// send above the dust limit (as the router can always overpay to meet the dust limit).
+		let mut remaining_msat_below_dust_exposure_limit = None;
+		let mut dust_exposure_dust_limit_msat = 0;
+
+		let (htlc_success_dust_limit, htlc_timeout_dust_limit) = if self.opt_anchors() {
+			(self.counterparty_dust_limit_satoshis, self.holder_dust_limit_satoshis)
+		} else {
+			let dust_buffer_feerate = self.get_dust_buffer_feerate(None) as u64;
+			(self.counterparty_dust_limit_satoshis + dust_buffer_feerate * htlc_success_tx_weight(false) / 1000,
+			 self.holder_dust_limit_satoshis       + dust_buffer_feerate * htlc_timeout_tx_weight(false) / 1000)
+		};
```

Comment:
We only implement the safer zero-fee HTLC tx variant of anchors.

Divide by a thousand because our the dust buffer federate is in terms of kw units

Reply:
> We only implement the safer zero-fee HTLC tx variant of anchors.

Ohh gotcha, thanks

> Divide by a thousand because the dust_buffer_feerate is in terms of kw units

Ah my bad I was thinking per_kw as (sats/1000wu) instead of (1000*sats/wu)